src/annoylib.h

```diff
@@ -827,21 +828,30 @@ template<typename S, typename T, typename Distance, typename Random>
     return item;
   }
 
-  void _get_all_nns(const T* v, size_t n, size_t search_k, vector<S>* result, vector<T>* distances) {
+  void _get_all_nns(const T* v, size_t n, size_t search_k, vector<S>* result, vector<T>* distances) const {
     Node* v_node = (Node *)malloc(_s); // TODO: avoid
     memcpy(v_node->v, v, sizeof(T)*_f);
     D::init_node(v_node, _f);
 
-    std::priority_queue<pair<T, S> > q;
+    typedef std::pair<T, S> qpair_t;
+    typedef std::vector<qpair_t> qvector_t;
+
+    size_t const roots_size = _roots.size();
 
     if (search_k == (size_t)-1)
-      search_k = n * _roots.size(); // slightly arbitrary default value
+      search_k = n * roots_size; // slightly arbitrary default value
+
+    qvector_t qvector;
+    qvector.reserve(roots_size); // prealloc queue
```

Comment:
could probably preallocate search_k items (not super important)

Reply:
fixed